roles/ceph-common/tasks/main.yml

```diff
@@ -0,0 +1,84 @@
+---
+- include: ./checks/check_system.yml
+
+- include: ./checks/check_mandatory_vars.yml
+
+- include: ./misc/system_tuning.yml
+  when: osd_group_name in group_names
+
+- include: ./pre_requisites/prerequisite_ice.yml
+  when: ceph_stable_ice
+
+- include: ./pre_requisites/prerequisite_rh_storage_iso_install.yml
+  when:
+    ceph_stable_rh_storage and
+    ceph_stable_rh_storage_iso_install
+
+- include: ./pre_requisites/prerequisite_rh_storage_cdn_install.yml
+  when:
+    ceph_stable_rh_storage and
+    ceph_stable_rh_storage_cdn_install
+
+- include: ./installs/install_on_redhat.yml
+  when: ansible_os_family == 'RedHat'
+
+- include: ./installs/install_on_debian.yml
+  when: ansible_os_family == 'Debian'
+
+- name: check for a ceph socket
+  shell: "stat /var/run/ceph/*.asok > /dev/null 2>&1"
+  changed_when: false
+  ignore_errors: true
+  register: socket
+
+- name: check for a rados gateway socket
+  shell: "stat {{ rbd_client_admin_socket_path }}*.asok > /dev/null 2>&1"
+  changed_when: false
+  ignore_errors: true
+  register: socketrgw
+
+- name: create a local fetch directory if it doesn't exist
+  local_action: file path=fetch state=directory 
+  changed_when: false
+  sudo: false
+
+- name: generate cluster uuid
+  local_action: shell uuidgen | tee fetch/ceph_cluster_uuid.conf
+    creates=fetch/ceph_cluster_uuid.conf
+  register: cluster_uuid
+  sudo: false
+  when: fsid != '4a158d27-f750-41d5-9e7f-26ce4c9d2d45'
+
+- name: read cluster uuid if it already exists
+  local_action: command cat fetch/ceph_cluster_uuid.conf
+    removes=fetch/ceph_cluster_uuid.conf
+  changed_when: false
+  register: cluster_uuid
+  sudo: false
+  when: fsid != '4a158d27-f750-41d5-9e7f-26ce4c9d2d45'
```

Comment:
This three task block seems very strange. I don't understand why uuidgen can't be ran and have the results registered. Dumping to a local file and then catting it just feels wrong.